### DISCUSSION TITLE
Active Directory Authentication Error

### DISCUSSION THREAD

**Discussion by mfulk-mk** [JanssenProject/jans#8525] [Q&A]
I had this working with version 1.0.22, but now setting up 1.1.1 attempting to login fails. From both CASA and the regular login screen I get the error "Failed to Authenticate". As far as I can tell, I have configured everything the same way. 

- JansLink connects and syncs with the Active Directory LDAP
- Added the same LDAP connector (in the TUI) under Auth Server > Authn with <Add Source LDAP Server> which successfully Tests.
- I do have CASA installed, but set the LDAP connector as the default for my client (however, I did try authenticating via CASA and noticed the same in the log)
Turned on Trace logging, I noticed this:

DEBUG [qtp1260134048-20] 98f60ad4-f1f1-4d5a-a1b3-399d7b777118 [jans.orm.ldap.operation.watch.OperationDurationUtil] (OperationDurationUtil.java:26) - LDAP operation: search, duration: PT0.114363482S, dn: DC=_COMPANY_,DC=_DOMAIN_, filter: (&(&(objectClass=jansPerson))(&(&(objectClass=top)(objectClass=jansPerson))(&(sAMAccountName=_username_)))), scope: SUB, batchOperationWraper: io.jans.orm.ldap.impl.LdapBatchOperationWraper@4c820f9c, start: 0, searchLimit: 100, count: 1, controls: null, attributes: null

It appears to be attempting to authenticate/connect to the LDAP with using the objectClass of jansPerson. But our LDAP should have the filter set to objectClass=person. Is this from the recently added jansPerson attribute from pull request #8183 or is there a place this filter is configured that I could not find?

**Comment by ossdhaval** [comment]
Adding @yurem to get his comments.

**Comment by ossdhaval** [comment]
@mfulk-mk Thank you for sharing this.

I would encourage you to implement the same scenario on latest and more stable releases. Let us know if this still is an issue. FYI, Jans Link is not actively developed component anymore. 

I am closing this discussion as outdated. Feel free to open a new one if you still need help.

**Comment by nynymike** [comment]
That search won't work against AD, because there are no entries with OC `jansPerson`... you'll need to figure out where to update the filter:  `(&(&(objectClass=jansPerson))(&(&(objectClass=top)(objectClass=jansPerson))(&(sAMAccountName=username))))` 

We halted work on Jans Link. It's a workforce application, and Janssen Server is used primarily for customer and citizen authentication and authorization. 

@ossdhaval is 100% right about version 1.0.22. And we halted work on Jans Link. The current version is 2.2.0.
